src/components/authorization/authorization.service.ts

```diff
@@ -104,6 +109,37 @@ export class AuthorizationService {
     return parseSecuredProperties(props, permissions, resource.SecuredProps);
   }
 
+  async verifyCanEditChanges<
+    TObject extends Resource,
+    Key extends keyof TObject & string
+  >(
+    baseNode: TObject,
+    props: readonly Key[],
+    changes: { [Key in keyof TObject]?: UnwrapSecured<TObject[Key]> }
+  ) {
+    for (const prop of props) {
```

Comment:
Part of my #1651 refactor was to prevent us from having to redefine prop lists everywhere. This same concept can be applied here as well. a.k.a. Let's pass in the `User` class instead of a list of prop keys.

Here's how that could work:
```suggestion
  async verifyCanEditChanges<Resource extends ResourceShape<any>>(
    resource: Resource,
    changes: Partial<UnsecuredDto<Resource['prototype']>>
  ) {
    for (const prop of resource.SecuredProps) {
```
`Resource` here is the "static" class, so we use `Resource['prototype']` to get a shape of the "instance" of the class.
Other than that, `changes` is the same type just written another way.

I think it's fine to verify all changes here. If we don't want that then the changes passed in can just be a subset from the caller.

Reply:
Now thinking about some edge cases like defined files. Maybe we don't need to assert the type of each field value.
```tsx
changes: Partial<Record<keyof Resource['prototype'], any>>
```
This would signify that the keys need to match but the values can be anything.